src/python/pants/backend/scala/lint/scalafix/rules.py

```diff
@@ -321,7 +325,14 @@ async def _run_scalafix_process(
                         else ()
                     ),
                     *(("--check",) if request.check_only else ()),
-                    *((f"--scalac-options={arg}" for arg in scalac.args) if scalac.args else ()),
+                    *(
```

Comment:
This could be
```
*(f"--scalac-options={arg}" for arg in (scalac.parsed_args_for_resolve(partition_info.resolve_name) or ()))
```

and then you can get rid of the temporary var entirely.